src/realm/object-store/sync/app.hpp

```diff
@@ -57,15 +57,28 @@ class App : public std::enable_shared_from_this<App>,
             public Subscribable<App> {
 public:
     struct Config {
+        // Information about the device where the app is running
+        struct DeviceInfo {
```

Comment:
can we just make this a map of std::string -> std::string? I don't think core need to be involved in precisely which fields the SDKs want to send to the server here.

Reply:
I had mentioned it in the Scope doc, but they didn't want to constrain the parameters to prevent different values being set by the different SDKs. The current set of 3 fixed parameters is not currently used consistently across the SDKs as it is...

Reply:
Make this a map.

Reply:
TBH, I don't like this. The fact that SDKs cannot be consistent in how they name some metadata fields  shouldn't mean that we have to bake those values into the core database. We are choosing to make our software more brittle for reasons that seem really dubious to me. That said, this is not a hot path, or even a path that I've ever had to worry about before, and I must pick my battles. 🚢 🇮🇹 😞

Reply:
I must be missing the reasons we wouldn't want to enforce this one place. I'm not aware any SDK need to diverge - on the contrary. The reason we have this difference now is that it was possible to diverge.
It seems to me that we are making the whole system less brittle by enforcing/codifying the contract we have with the server (aka making it "type-safe"). 
What am I missing?

Reply:
Any conclusions regarding how we want to approach this?

Reply:
The Object Store part of Core is all about bringing every common functionality and behaviour into a common place implemented in C++. It's not the generic database that is opaque and flexible. The project was established for the specific purpose to unify these parameters, and it's kind of a "fixed" contract what to inform about. I'm happy to be proved wrong if I've misunderstood something? But I think I'm just repeating myself here, and perhaps it's better to ask a few of the SDK people that will use it? @nirinchev @cmelchior

Reply:
Agree with Brian here. The idea behind moving all of this to ObjectStore was to prevent SDK's from making mistakes when sending connection metadata. So turning this into a generic `Map<String, String>` would seem to go against that.